pkg/container-collection/k8s.go

```diff
@@ -53,6 +57,12 @@ func NewK8sClient(nodeName string) (*K8sClient, error) {
 
 	fieldSelector := fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
 
+	kubeletConfig, err := getCurrentKubeletConfig(clientset, nodeName)
```

Comment:
This method can return nil in kubeletConfig, currently you're only logging the error which will cause a runtime error in line 65 in that case when you do: `kubeletConfig.ContainerRuntimeEndpoint`
How do you want to handle that case?

Reply:
Returning an error message upon failure